src/node_interface.rs

```diff
@@ -40,7 +50,7 @@ pub trait Interface : Sync + Send {
                       from_authority: Authority,
                       from_address: NameType) -> Result<MessageAction, InterfaceError>;
 
-    /// if reply is data then we send back the response message (ie get_response )
+    /// data or address of the node potentially storing data is returned on success.
```

Comment:
Correction - ignore above: document with 'should' and refer to persona with a provided authority type.  The message action on Reply(data) (can) contain data to be sent back to the original requestor; error will go back to the previous source (one hop back); SendOn sends the message to a new destination, with us as source